src/Bicep.Core/Syntax/SyntaxRewriteVisitor.cs

```diff
@@ -373,7 +373,7 @@ protected virtual SyntaxBase ReplaceIntegerLiteralSyntax(IntegerLiteralSyntax sy
                 return syntax;
             }
 
-            return new IntegerLiteralSyntax(literal, long.Parse(literal.Text));
+            return new IntegerLiteralSyntax(literal, UInt64.Parse(literal.Text));
```

Comment:
[nit] For consistency, it would be good to either use `ulong` / `long`, or use `UInt64` / `Int64` - they're generally interchangeable.

Here I think you should be able to use `ulong.Parse` instead of `UInt64.Parse` if you like